### PR TITLE
Migrate to PNPM

### PR DESCRIPTION
**Pull request by jelhan**
Migrates the project from yarn v1 to PNPM. PNPM is the recommended package manager for the adopted-ember-addons org per https://github.com/adopted-ember-addons/program-guidelines/issues/28.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -49,14 +49,15 @@
   },
   "dependencies": {
     "broccoli-funnel": "^3.0.2",
+    "ember-auto-import": "^2.6.3",
```
Comment (jelhan) [inline package.json:52]: I moved this breaking change out to #553 which should be merged first.

File: package.json [hunk 2]
```diff
@@ -49,14 +49,15 @@
...
     "ember-cli-babel": "^7.26.11",
     "ember-cli-typescript": "^5.1.1"
   },
   "devDependencies": {
     "@babel/eslint-parser": "^7.21.3",
     "@babel/plugin-proposal-decorators": "^7.21.0",
     "@ember/optional-features": "^2.0.0",
-    "@ember/string": "^3.0.1",
+    "@ember/string": "^4.0.0",
```
Comment (jelhan) [inline package.json:60]: This fixes an issue introduced in #547. The project had a direct dependency on `^3.0.1` of `@ember/string` but a resolution forcing all transitive dependencies to resolve to `^4.0.0`. Please let me know if you prefer this to be pulled out to a separate PR.

**Review by SergeAstapov — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -19,17 +19,20 @@ jobs:
 
     steps:
       - uses: actions/checkout@v3
+      - uses: pnpm/action-setup@v2
+        with:
+          version: 8
```
Comment (SergeAstapov) [inline .github/workflows/ci.yml:24]: @jelhan out of curiosity, why not using pnpm v9 right away?

Comment (jelhan) [reply .github/workflows/ci.yml:24]: I followed the setup of the latest Ember CLI blueprints: https://github.com/ember-cli/ember-cli/blob/6691ea1008dd734a9c18c0ec5ed7a389987f608b/blueprints/addon/files/.github/workflows/ci.yml#L22-L24
